File: FashionFlows.BuildingBlock/FashionFlows.BuildingBlock.Shared/Events/OrderFailedEvent.cs
using FashionFlows.BuildingBlock.Domain.Events.Interface;
using FashionFlows.BuildingBlock.Domain.Model;

namespace FashionFlows.BuildingBlock.Domain.Events;

public class OrderFailedEvent : IOrderFailedEvent
{
    public Guid OrderId { get; set; }
    public Guid UserId { get; set; }
    public List<OrderItem> OrderItemList { get; set; }
}

File: FashionFlows.Services/Notification/internal/model/events/order_failed_event.go
package events

type OrderItemEvent struct {
	ProductID string `json:"productId"`
	Quantity  int    `json:"quantity"`
}

type OrderFailedEvent struct {
	OrderID       string           `json:"orderId"`
	UserID        string           `json:"userId"`
	OrderItemList []OrderItemEvent `json:"orderItemList"`
}
